src/raw/aws/fargate_inspector.go

```diff
@@ -129,10 +129,24 @@ func containerResponseToDocker(container ContainerResponse) types.Container {
 
 func processFargateLabels(labels map[string]string) map[string]string {
 	for label, value := range labels {
-		// The cluster name label has to be processed because in ECS with EC2 launch type it's only the cluster name.
-		// Meanwhile, in ECS with Fargate launch type the same label has the full cluster ARN as value.
-		if label == "com.amazonaws.ecs.cluster" && isECSARN(value) {
-			labels[label] = ecsClusterNameFromARN(value)
+		switch label {
+		case "com.amazonaws.ecs.cluster":
+			if isECSARN(value) {
+				// The cluster name label has to be processed because in ECS with EC2 launch type it's only the cluster name.
+				// Meanwhile, in ECS with Fargate launch type the same label has the full cluster ARN as value.
+				labels[label] = ecsClusterNameFromARN(value)
+				// preserve the original arn is preserved in the following synthetic label
+				labels["com.newrelic.nri-docker.cluster-arn"] = value
+				// Add label signaling fargate launch type
+				labels["com.newrelic.nri-docker.launch-type"] = "fargate"
```

Comment:
`launchType` will be set to `fargate` if the `com.amazonaws.ecs.cluster` label has an ARN format. This is somehow a "weak" indicator of whether we are running in Fargate or not.

If we can guarantee that the agent/backend will override this setting, we should be fine with this.

Reply:
On secondary inspection, I found out that this is actually cannot happen, since far label processing is done only when the `FARGATE` env is set.